✨ feat: Add Keycloak SSO provider support

0xff26b9a8 opened the pull request:
- Implemented Keycloak authentication flow
- Added configuration options for Keycloak server
- Updated documentation with setup instructions

#### 💻 变更类型 | Change Type

<!-- For change type, change [ ] to [x]. -->

- [x] ✨ feat
- [ ] 🐛 fix
- [ ] ♻️ refactor
- [ ] 💄 style
- [ ] 👷 build
- [ ] ⚡️ perf
- [x] 📝 docs
- [ ] 🔨 chore

#### 🔀 变更说明 | Description of Change

支持 Keycloak 提供商，增加了相对应字段，添加了文档说明
<!-- Thank you for your Pull Request. Please provide a description above. -->

#### 📝 补充信息 | Additional Information

<!-- Add any other context about the Pull Request here. -->

0xff26b9a8 committed:
✨ feat: Add Keycloak SSO provider support

- Implemented Keycloak authentication flow
- Added configuration options for Keycloak server

arvinxx commented on src/config/auth.ts:
新增的 SSO Provider 不需要再在这里添加了 env 了

arvinxx commented on src/libs/next-auth/sso-providers/keycloak.ts:
```diff
@@ -0,0 +1,29 @@
+    // Specify auth scope, at least include 'openid email'
+    authorization: { params: { scope: 'openid email profile' } },
+    // TODO(NextAuth ENVs Migration): Remove once nextauth envs migration time end
+    clientId: authEnv.KEYCLOAK_CLIENT_ID ?? process.env.AUTH_KEYCLOAK_ID,
```
这里应该直接用 `AUTH_KEYCLOAK_XXX` 就好了

0xff26b9a8 replied:
好的，了解了，我修改一下然后增加一下文档

arvinxx reviewed (changes requested):
实现修正一下就好，另外能麻烦补充下文档吗~

0xff26b9a8 committed:
🔥 chore: remove unused environment variables

- Removed deprecated configuration options

0xff26b9a8 committed:
📝 docs: add documentation for Keycloak SSO provider

- Added setup instructions for Keycloak integration
- Documented required environment variables
- Updated main SSO documentation to include Keycloak option

0xff26b9a8 commented:
> 实现修正一下就好，另外能麻烦补充下文档吗~

麻烦看一下，已经修正了env，然后添加了文档，不确定还有没有其他地方需要增加文档，有的话和我说一下

arvinxx commented on docs/self-hosting/advanced/auth/next-auth/keycloak.zh-CN.mdx:
```diff
@@ -0,0 +1,116 @@
+- 你私有部署 Keycloak，其域名为 `https://lobe-auth-api.example.com`。
+
+<Steps>
+  &#x20; \### 创建 Keycloak 领域和客户端
```
这里的符号是不是有问题？

0xff26b9a8 replied:
？？？，发现了，我这是在本地写完然后通过todesk复制到远程的机器上，然后变成了这样，离谱

arvinxx commented on docs/self-hosting/advanced/auth/next-auth/keycloak.mdx:
这里也有问题

arvinxx commented on src/libs/next-auth/sso-providers/keycloak.ts:
```diff
@@ -0,0 +1,27 @@
+    ...CommonProviderConfig,
+    // Specify auth scope, at least include 'openid email'
+    authorization: { params: { scope: 'openid email profile' } },
+    // TODO(NextAuth ENVs Migration): Remove once nextauth envs migration time end
```
TODO 可以删掉了，包括下面的 remove end

0xff26b9a8 committed:
🔍 fix: correct documentation issues and remove redundant comments

- Fixed formatting in Keycloak SSO documentation
- Removed outdated comments in authentication code
